src/ydata_synthetic/synthesizers/regular/cgan/model.py

```diff
@@ -172,8 +172,8 @@ def __init__(self, batch_size, num_classes):
     def build_model(self, input_shape, dim, data_dim):
         noise = Input(shape=input_shape, batch_size=self.batch_size)
         label = Input(shape=(self.num_classes,), batch_size=self.batch_size, dtype='int32')
-        label_embedding = Embedding(2, 1)(label)
-        input = Flatten()(multiply([noise, label_embedding]))
+        label_dense = Dense(1)(label)
+        input = multiply([noise, label_dense])
```

Comment:
Flatten was no longer required since no extra dimensions are added to the processed label.